### PR TITLE
Remove the dataspaces cache feature

### PR DESCRIPTION
**Pull request by paraita**
To be merged with https://github.com/ow2-proactive/studio/pull/284 if this PR is merged.

### CODE REVIEW

**Comment by paraita** [comment]
retest this please

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 1]
```diff
@@ -70,39 +66,34 @@
...
     private boolean linuxOS;
 
-    private static transient ReentrantLock cacheTransferLock = new ReentrantLock();
-    private SpaceInstanceInfo cacheSpaceInstanceInfo;
-
-    private transient StringBuffer clientLogs = new StringBuffer();
+    private StringBuffer clientLogs = new StringBuffer();
 
-    private transient ExecutorService executorTransfer =
+    private ExecutorService executorTransfer =
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: CRITICAL Make "executorTransfer" transient or serializable. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1948)

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 2]
```diff
@@ -70,39 +66,34 @@
...
     public static final String PA_NODE_DATASPACE_CREATE_FOLDER_HIERARCHY_SEQUENTIALLY =
             "pa.node.dataspace.create_folder_hierarchy_sequentially";
 
-    private transient DataSpacesFileObject SCRATCH;
-    private transient DataSpacesFileObject CACHE;
-    private transient DataSpacesFileObject INPUT;
-    private transient DataSpacesFileObject OUTPUT;
-    private transient DataSpacesFileObject GLOBAL;
-    private transient DataSpacesFileObject USER;
+    private DataSpacesFileObject SCRATCH;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: CRITICAL Make "SCRATCH" transient or serializable. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1948)

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 3]
```diff
@@ -70,39 +66,34 @@
...
+    private DataSpacesFileObject INPUT;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: CRITICAL Make "INPUT" transient or serializable. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1948)

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 4]
```diff
@@ -70,39 +66,34 @@
...
+    private DataSpacesFileObject OUTPUT;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: CRITICAL Make "OUTPUT" transient or serializable. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1948)

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 5]
```diff
@@ -70,39 +66,34 @@
...
+    private DataSpacesFileObject GLOBAL;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: CRITICAL Make "GLOBAL" transient or serializable. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1948)

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 6]
```diff
@@ -70,39 +66,34 @@
...
+    private DataSpacesFileObject USER;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: CRITICAL Make "USER" transient or serializable. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1948)

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 7]
```diff
@@ -70,39 +66,34 @@
...
 
     private TaskId taskId;
-    private transient NamingService namingService;
+    private NamingService namingService;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: CRITICAL Make "namingService" transient or serializable. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1948)

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContext.java [hunk 8]
```diff
@@ -60,29 +60,40 @@
...
-                nodeDataSpacesURIs, progressFilePath, currentNodeHostname, null);
+            TaskResult[] previousTasksResults, String scratchURI, String inputURI, String outputURI,
+            String userURI,
+            String globalURI, String progressFilePath, String currentNodeHostname) throws NodeException {
+        this(executableContainer, initializer, previousTasksResults, scratchURI, inputURI, outputURI, userURI,
+                globalURI, progressFilePath, currentNodeHostname, null);
     }
 
 
     public TaskContext(ExecutableContainer executableContainer, TaskLauncherInitializer initializer,
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContext.java]: MAJOR Constructor has 11 parameters, which is greater than 7 authorized. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00107)

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContext.java [hunk 9]
```diff
@@ -60,29 +60,40 @@
...
+    private final String inputURI;
+    private final String outputURI;
+    private final String userURI;
+    private final String globalURI;
     private final String schedulerHome;
 
     private final String progressFilePath;
 
 
     public TaskContext(ExecutableContainer executableContainer, TaskLauncherInitializer initializer,
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContext.java]: MAJOR Constructor has 10 parameters, which is greater than 7 authorized. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00107)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 9 issues
* CRITICAL 7 critical
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by lpellegr** [comment]
Retest this please.

**Comment by lpellegr** [comment]
To be merged with https://github.com/ow2-proactive/studio/pull/284 if this PR is merged.

**Comment by paraita** [comment]
This PR is not relevant anymore as the instabilities in the tests have been taken care of in another PR.

**Pull request closed without merging**